Sources/NIO/ByteBuffer-views.swift

```diff
@@ -56,24 +56,22 @@ public struct ByteBufferView: RandomAccessCollection {
 
     public subscript(position: Index) -> UInt8 {
         get {
-            guard position >= self._range.lowerBound && position < self._range.upperBound else {
-                preconditionFailure("index \(position) out of range")
-            }
+            self._failEarlyRangeCheck(position, bounds: self._range)
```

Comment:
I don't think you're supposed to _call_ those underscored methods? I think you're just meant to implement them. Am I wrong here?

Reply:
CC @lorenty

Reply:
@weissi Can't we use underscored methods? I read collection classes in stblib and they often use `_failEarlyRangeCheck`  in `subscript` method. I thought it's ok so I replaced guard statement with `_failEarlyRangeCheck` to remove duplicate code.

Reply:
@trungducc I think the underscored methods are meant to be internal to the standard library and are not supposed to be called by users outside the standard library, @lorentey to confirm this

Reply:
The underscored functions are usually for some functionality the stdlib needs but no one is sure yet how to best expose it to the language, or whether we want users doing it at all. So they haven't gone through Swift evolution process, and aren't part of the API. They need to be in the ABI so that inlined stdlib functions can still cal them. It's too bad we at least don't have a warning for calling them outside the stdlib. Maybe @lorentey or @airspeedswift  can explain more. If it's something you think non-stdlib code needs and isn't easily reimplemented outside the stdlib, then someone should create an evolution proposal.